src/generator.cs

```diff
@@ -1918,6 +1938,222 @@ public static bool HasForcedAttribute (ICustomAttributeProvider cu, out string o
 		return true;
 	}
 
+	public static bool HasBindAsAttribute (ICustomAttributeProvider cu, out Type bindAs, out bool isNullable)
+	{
+		var att = GetAttribute<BindAsAttribute> (cu) ?? GetAttribute<BindAsAttribute> ((cu as MethodInfo)?.ReturnParameter);
+
+		if (att == null) {
+			bindAs = null;
+			isNullable = false;
+			return false;
+		}
+
+		bindAs = att.Type;
+		isNullable = att.IsNullable;
+		return true;
+	}
+
+	void GetBindAsSetterWrapper (MemberInformation minfo, ref string args)
```

Comment:
If I understand this correctly, you have two Get*Wrapper methods to cover converting both ways, either to the BindAs type, or from it.

If that's right, it might be better to name the methods `GetFromBindAsWrapper`/`GetToBindAsWrapper` instead, where one is used for setters and input parameters in methods, and the other for getters and return values from methods.